File: algorithms/04_java/math/Catalan.java
public final class Catalan {
    private Catalan() {}

    public static void main(String[] args) {
        long[] catalan = new long[11];
        catalan[0] = 1;
        for (int i = 1; i <= 10; i++) {
            long sum = 0;
            for (int j = 0; j < i; j++) {
                sum += catalan[j] * catalan[i - 1 - j];
            }
            catalan[i] = sum;
        }
        if (catalan[5] != 42 || catalan[10] != 16796) {
            throw new AssertionError("wrong Catalan numbers");
        }
        System.out.println("catalan(10)=" + catalan[10]);
    }
}

File: algorithms/15_ruby/math/catalan.rb
# Catalan numbers by the recurrence.
catalan = Array.new(11, 0)
catalan[0] = 1
(1..10).each do |i|
  sum = 0
  (0...i).each { |j| sum += catalan[j] * catalan[i - 1 - j] }
  catalan[i] = sum
end
raise "wrong Catalan numbers" unless catalan[5] == 42 && catalan[10] == 16_796

puts "catalan(10)=#{catalan[10]}"
